### DISCUSSION TITLE
Get list of a (any) entity, using GID array

### DISCUSSION THREAD

**Discussion by miksony** [Shopify/storefront-api-feedback#202] [Show & Tell]
Hi everyone 👋

I was able to retrieve a list of Product nodes and Collection nodes only by using a global id array.
Probably this can be done on any other entity.
I found that query very useful for some of the cases I needed.

```
query getProducts($ids: [ID!]!) {
    nodes(ids: $ids) {
        ...on Product {
          id
          title
        }
   }
}
```

this is just a simple query which is returning something like this

```
{
	"data": {
		"nodes": [
			{
				"id": "gid://shopify/Product/8555747901741",
				"title": "Soda 2"
			},
			{
				"id": "gid://shopify/Product/8555873337645",
				"title": "Soda single 3"
			}
		]
	}
}
```

input for that query is "ids" array which is contained of **BASE64 encoded global ids**, so in my case something like this

```
{
	"ids": [
		"Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzg1NTU3NDc5MDE3NDE=",
		"Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzg1NTU4NzMzMzc2NDU="
	]
}
```

**Comment by johncraigcole** [comment]
Thanks for sharing this with the community @miksony. Yes this will work with any type the Storefront API supports. One thing I wanted to share is that you no longer have to BASE64 encode ids. There's a developer change log entry on the subject you can review [here](https://shopify.dev/changelog/changes-to-storefront-api-object-ids#:~:text=Last%20year%2C%20we%20announced%20that,no%20longer%20be%20Base64%20encoded.)
